### ISSUE TITLE
Check if level is active

### ISSUE THREAD

**Issue by climbeault** [Trilliant/SLF4EC#2]
It would be nice to have an interface method/macro to check if a level is currently active for a log.
This would allow an application to conditionally execute code in preparation for a log output.
As an example:

``` C
if (logIsActive(Network, LEVEL_TRACE))
{
  // Format packet in preparation for logging.
  // ...
  logTrace(Network, "Received packet: %s", formattedPacket);
}
```

**Comment by jeremfg** [comment]
If I add the following macro to log.h, does it satisfy your request?

``` C
/**
 * Checks if logging is active for the specified @p logCategory and @p logLevel.
 *
 * @param[in] logCategory ::LogCategory to check
 * @param[in] logLevel LogLevel to check
 *
 * @retval true if the @p logLevel is active for the given @p logCategory.
 * @retval false if the @p logLevel is NOT active for the given @p category.
 */
#define logIsActive(logCategory, logLevel) \
    (logCategory.currentLogLevel <= logLevel)
```

By my understanding, you were only looking for a prettier way of getting the answer you could already get by making the comparison yourself. Am I correct?

**Comment by climbeault** [comment]
That's right.
This is prettier but also, the client code doesn't have to take into account the ordering of the levels by using the macro.

However, unless I am wrong, I think it should be >= for the comparison, as used in isCategoryActive() function. For better isolation of level ordering dependency, the comparison should appear only at one place.

**Comment by jeremfg** [comment]
You are correct for the comparison

**Comment by climbeault** [comment]
I just realized that the `logIsActive` macro should be written as following since `logCategory` may be an expression (a dereferenced pointer would fail otherwise).

``` C
#define logIsActive(logCategory, logLevel) \
    ((logCategory).currentLogLevel >= logLevel)
```

**Comment by jeremfg** [comment]
Agreed

**Comment by jeremfg** [comment]
FIxed in 1.3.0
